src/testers.jl

```diff
@@ -0,0 +1,180 @@
+function _ensure_not_running_on_functor(f, name)
```

Comment:
I am tempted to break this up futher but I am not sure how.
Maybe
`/testers/common.jl`, `/testers/frule.jl`, `/testers/rrule.jl`, `/testers/scalarrule.jl` 
?

Reply:
this seems good for now

I think being able to see all these in one place might be helpful for refactor work that's likely to happen